src/controllers/userController.ts

```diff
@@ -2,36 +2,247 @@ import { Request, Response } from 'express';
+      where: {
+        id,
+      },
+      attributes: ['id', 'firstName', 'lastName', 'email', 'phoneNumber', 'photoUrl', 'gender'],
+      include: {
+        model: Role,
+        attributes: ['name'],
+      },
+    });
+    if (!user) return res.status(404).json({ ok: false, err: 'User with this ID does not exits' });
+
+    return res.status(200).json({ ok: false, message: user });
+  } catch (error) {
+    logger.error(`Error for get one user: ${error}`);
+    sendInternalErrorResponse(res, error);
+    return;
+  }
+};
+// Function for delete a user
+export const deleteUser = async (req: Request, res: Response) => {
+  try {
+    const deleteUserAccount = await User.destroy({
+      where: {
+        id: req.params.id,
+      },
+    });
+    if (deleteUserAccount === 0) {
+      return res.status(404).json({ ok: false, message: 'User with this ID does not exist' });
+    }
+    return res.status(200).json({ ok: true, message: 'User deleted successfully' });
+  } catch (error) {
+    logger.error('Error deleting user: ', error);
+    sendInternalErrorResponse(res, error);
+    return;
+  }
+};
+// Function to edit user role
+export const editUserRole = async (req: Request, res: Response) => {
+  try {
+    const { userId } = req.params;
+    const { roleId } = req.body;
+    if (validateFields(req, ['roleId']).length !== 0) {
+      res.status(400).json({ ok: false, error: 'roleId is required' });
+      return;
+    }
+    const user = await User.findByPk(userId);
+
+    if (!user) {
+      logger.error('Error User Not Found');
+      res.status(404).json({ ok: false, error: 'User not found' });
+    }
+
+    await user?.update({ RoleId: roleId });
+
+    res.status(200).json({ ok: true, message: 'Role assigned successfully.' });
+  } catch (error) {
+    logger.error('Error Edit User Role: ', error);
+    sendInternalErrorResponse(res, error);
   }
 };
+// Function to update user profile
+export const editUser = async (req: Request, res: Response) => {
+  try {
+    const { firstName, lastName, gender, phoneNumber } = req.body;
+
+    const requiredFields = ['firstName', 'lastName', 'gender', 'phoneNumber'];
+    const missingFields = validateFields(req, requiredFields);
+
+    if (missingFields.length > 0) {
+      logger.error(`Required fields are missing:${missingFields.join(', ')}`);
+      res.status(400).json({
+        ok: false,
+        message: `Required fields are missing: ${missingFields.join(', ')}`,
+      });
+      return;
+    }
+
+    if (req.body.email || req.body.password) {
+      res.status(403).json({ ok: false, message: 'It does not possible to update email and password here' });
+    }
+    // when autherization applyed i will be able to access req.user
+    // const user: User | null = await User.findByPk(req.user.dataValues.id) // uncomment when auth middleware
+    const user = await User.findOne({ where: { id: req.params.id } }); // delete this after auth middleware a
```

Comment:
Let it stay like this for the purpose of allowing the admin to update this data. This time around, the ID will belong to the admin; therefore, you understand that they will need the ID of the user they are updating from the params. This is okay—leave it as it is and indeed remove that comment